src/Messages/Interactions/TSMessage.h

```diff
@@ -60,6 +64,11 @@ typedef NS_ENUM(NSInteger, TSGroupMetaMessage) {
 
 - (BOOL)hasAttachments;
 
+// This message should return TSMessage.receivedAtDate for most messages.
+// For messages received before TSMessage.receivedAtDate was added, this
+// will try to return TSIncomingMessage.receivedAt.
+- (nullable NSDate *)bestReceivedAtDate;
```

Comment:
nit: I'm having a hard time keeping track of the 3 similar-but-different fields in the inheritance hierarchy:

2. TSIncomingMessage.receivedAt (local time when message was received)
3. TSMessage.receivedAtDate
4. TSMessage.bestReceivedAtDate

Since we don't really "Receive" outgoing messages (except synced transcripts), and since it's only "best" in the context of sorting, what do you think of calling it something like `sortingDate` to differentiate it from the others?